File: backend/app/schemas/subject.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class SubjectBase(BaseModel):
    name: str
    is_active: bool = True


class SubjectCreate(BaseModel):
    name: str


class SubjectRead(SubjectBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class SubjectUpdate(BaseModel):
    name: str | None = None
    is_active: bool | None = None

File: frontend/src/types/subject.ts
// Mirrors backend/app/schemas/subject.py in full.

export interface Subject {
  id: number;
  name: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SubjectCreate {
  name: string;
}

export interface SubjectUpdate {
  name?: string;
  is_active?: boolean;
}
